.github/workflows/rust.yaml

```diff
@@ -71,4 +71,4 @@ jobs:
       - uses: actions-rs/cargo@v1
         with:
           command: clippy
-          args: --all-targets --all-features -- -D warnings
```

Comment:
This is opening the door for all the warnings. Today have none due to this. 

Are we sure we want this?

Reply:
The alternative is to pin `rustc` to `1.80`, the one that is marking the allowing of `box_pointers` is `rustc 1.81.0 (eeb90cda1 2024-09-04)`

Reply:
What about  updating `src/envoy` path content? I am aware it is auto-generated from protobuf descriptors. So maybe change the source of truth or add some customization https://github.com/Kuadrant/wasm-shim/blob/main/build.rs#L55-L106

It is all about deleting 

```rust
#![allow(box_pointers)]
```

That is a lint being removed in rust 1.80